libs/build_met.sh

```diff
@@ -0,0 +1,136 @@
+    module load hdf5
+    module load netcdf
+    module load bufr
+    module load g2c
+    module load zlib
+    module load jasper
+    module load png
+    module load gsl
+    module list
+    set -x
+    prefix="${PREFIX:-"/opt/modules"}/$compiler/$mpi/$name/$version"
+    if [[ -d $prefix ]]; then
+	[[ $OVERWRITE =~ [yYtT] ]] && ( echo "WARNING: $prefix EXISTS: OVERWRITING!";$SUDO rm -rf $prefix ) \
+	    || ( echo "WARNING: $prefix EXISTS, SKIPPING"; exit 1 )
+    fi
+else
+
+    prefix=${MET_ROOT:-"/usr/local"}
+
+fi
+
+export MET_BASE=$prefix/share/met
+
+if [[ ! -z $mpi ]]; then
+    export FC=$MPI_FC
+    export CC=$MPI_CC
+    export CXX=$MPI_CXX
+else
+    export FC=$SERIAL_FC
+    export CC=$SERIAL_CC
+    export CXX=$SERIAL_CXX
+fi
+
+export F77=$FC
+export FFLAGS="${STACK_FFLAGS:-} ${STACK_met_FFLAGS:-}"
+export CFLAGS="${STACK_CFLAGS:-} ${STACK_met_CFLAGS:-}"
+
+export MET_NETCDF=${NetCDF_ROOT}
+export MET_HDF5=${HDF5_ROOT}
+export MET_BUFRLIB=${bufr_ROOT}/lib
+export MET_GRIB2CLIB=${g2c_ROOT}/lib
+export MET_GRIB2CINC=${g2c_ROOT}/include
+export MET_GSL=${GSL_ROOT}
+export BUFRLIB_NAME=-lbufr_4
+export GRIB2CLIB_NAME=-lg2c
+export LIB_JASPER=${JASPER_ROOT}/lib64
+export LIB_LIBPNG=${PNG_ROOT}/lib64
+export LIB_Z=${ZLIB_ROOT}/lib
+
+LDFLAGS1="-Wl,--disable-new-dtags"
+LDFLAGS2="-Wl,-rpath,${MET_NETCDF}/lib:${MET_HDF5}/lib:${MET_BUFRLIB}"
+LDFLAGS3="-Wl,-rpath,${MET_GRIB2CLIB}:${MET_PYTHON}/lib:${MET_GSL}/lib"
+LDFLAGS4="-L${LIB_JASPER} -L${MET_HDF5}/lib -L${LIB_LIBPNG} -L${LIB_Z}"
+LDFLAGS5="-L${I_MPI_ROOT}/lib64"
+if [[ -z $mpi ]]; then
+  export LDFLAGS="${LDFLAGS1:-} ${LDFLAGS2:-} ${LDFLAGS3:-} ${LDFLAGS4:-}"
+else
+  export LDFLAGS="${LDFLAGS1:-} ${LDFLAGS2:-} ${LDFLAGS3:-} ${LDFLAGS4:-} ${LDFLAGS5:-}"
+fi
+  export LIBS="-lhdf5_hl -lhdf5 -lz"
+
+
+export CFLAGS+="-D__64BIT__"
+export CXXFLAGS+="-D__64BIT__"
+
+cd ${HPC_STACK_ROOT}/${PKGDIR:-"pkg"}/${pkg_name}
+curr_dir=$(pwd)
+echo "$(pwd)"
+
+echo "MET Configuration settings..."
+printenv | egrep "^MET_" | sed -r 's/^/export /g'
+echo "LDFLAGS = ${LDFLAGS}"
+
+echo "./configure --prefix=$prefix BUFRLIB_NAME=${BUFRLIB_NAME} GRIB2CLIB_NAME=${GRIB2CLIB_NAME} --enable-grib2 --enable-python"
+./configure --prefix=$prefix BUFRLIB_NAME=${BUFRLIB_NAME} GRIB2CLIB_NAME=${GRIB2CLIB_NAME} --enable-grib2 --enable-python
+
+ret=$?
+if [ $ret != 0 ]; then
+    echo "configure returned with non-zero ($ret) status"
+    exit 1
+fi
+
+echo "make > make.log 2>&1"
+make > make.log 2>&1
+ret=$?
+if [ $ret != 0 ]; then
+    echo "make returned with non-zero ($ret) status"
+    exit 1
+fi
+
+echo "make install > make_install.log 2>&1"
+make install > make_install.log 2>&1
+ret=$?
+if [ $? != 0 ]; then
+    echo "make install returned with non-zero ($ret) status"
+    exit 1
+fi
+
+echo "make test > make_test.log 2>&1"
+make test > make_test.log 2>&1
+ret=$?
+if [ $? != 0 ]; then
+    echo "make test returned with non-zero ($ret) status"
+    exit 1
+fi
```

Comment:
This should be wrapped in a if block of `MAKE_CHECK`.

Reply:
@aerorahul 
I'm not exactly sure what you mean when you say that 
> This should be wrapped in a if block of MAKE_CHECK.
I am not familiar with how this works.  If I grep for MAKE_CHECK in the files, I see that some have multiple lines with MAKE_CHECK in them and some only have one line:
```
[Julie.Prestopnik@m71a1 libs]$ grep -i MAKE_CHECK *
build_atlas.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && ctest
build_cdo.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_eckit.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && ctest
build_eigen.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_esmf.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_esmf.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make installcheck
build_fckit.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && ctest
build_fftw.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_fms.sh:#[[ $MAKE_CHECK =~ [yYtT] ]] && make check # make check is not implemented in cmake builds
build_gptl.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_gsl_lite.sh:#[[ $MAKE_CHECK =~ [yYtT] ]] && make test
build_gsl.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_hdf5.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_jasper.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_jpeg.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && check_flags="-DBUILD_TESTS=ON"
build_jpeg.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make test
build_json_schema_validator.sh:      -DBUILD_TESTS=$MAKE_CHECK \
build_json_schema_validator.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make test
build_json.sh:      -DJSON_BuildTests=$MAKE_CHECK
build_json.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make test
build_lapack.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_mpi.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_nccmp.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_nceplibs.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_nco.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_netcdf.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_netcdf.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_netcdf.sh:  [[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_pio.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_pnetcdf.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_png.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_szip.sh:[[ $MAKE_CHECK =~ [yYtT] ]] && make check
build_udunits.sh:[[ "$MAKE_CHECK" = "YES" ]] && make check
build_zlib.sh:[[ "$MAKE_CHECK" = "YES" ]] && make check
```
and not all of the lines are the same.  Can you please provide more specific guidance?

Reply:
@aerorahul 
I added "[[ $MAKE_CHECK =~ [yYtT] ]] && make check" above the "make test" line.  Perhaps that is all that is needed?